homework/index.js

```diff
@@ -30,14 +30,93 @@
+      const trName = createAndAppend('tr', tbody);
+      createAndAppend('th', trName, { text: 'Repository:' });
+      const tdName = createAndAppend('td', trName);
+      createAndAppend('a', tdName, { text: repo.name, href: repo.html_url, target: '_blank' });
+      const trDes = createAndAppend('tr', tbody);
+      createAndAppend('th', trDes, { text: 'Description:' });
+      createAndAppend('td', trDes, { text: repo.description });
+      const trForks = createAndAppend('tr', tbody);
+      createAndAppend('th', trForks, { text: 'Forks:' });
+      createAndAppend('td', trForks, { text: repo.forks_count });
+      const trUpd = createAndAppend('tr', tbody);
+      createAndAppend('th', trUpd, { text: 'Updated at:' });
+      createAndAppend('td', trUpd, { text: repo.updated_at });
+    });
+  }
+
+  function renderContributors(repository, ul) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repository}/contributors`;
+    fetchJSON(url, (err, contributions) => {
+      if (err) return;
+      ul.innerHTML = '<h3>Contributions</h3>';
+      contributions.forEach(contributor => {
+        const li = createAndAppend('li', ul, { class: 'contributor' });
+        const contributorTable = createAndAppend('table', li);
+        const contributorTbody = createAndAppend('tbody', contributorTable);
+        const contributorTr = createAndAppend('tr', contributorTbody);
+        const imgTd = createAndAppend('td', contributorTr);
+        createAndAppend('img', imgTd, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        const contributorName = createAndAppend('td', contributorTr);
+        const contributorNameH4 = createAndAppend('h4', contributorName);
+        createAndAppend('a', contributorNameH4, {
+          text: contributor.login,
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        createAndAppend('td', contributorTr, { text: contributor.contributions, class: 'num' });
+      });
+    });
+  }
+
   function main(url) {
     fetchJSON(url, (err, repositories) => {
       const root = document.getElementById('root');
+      const header = createAndAppend('header', root, { class: 'header' });
+      createAndAppend('h1', header, { text: 'HYF Repositories' });
+      const select = createAndAppend('select', header, { class: 'select' });
+      //  left side
+      const div = createAndAppend('div', root, { class: 'info-box' });
+      const ul = createAndAppend('ul', div, { class: 'info-ul' });
+      // right side
+      const right = createAndAppend('div', root, { class: ' contributions' });
+      const ulContributions = createAndAppend('ul', right, { class: 'right-ul' });
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
```

Comment:
It is best to sort the repositories at this point, as soon as you have established that there was no error.


In your event handler, as I mentioned elsewhere, you can then use `repositories[select.value]` to get the select repository object.